p2p/sql_stmt.go

```diff
@@ -2,36 +2,58 @@ package p2p
 
 import (
 	"database/sql"
+	"fmt"
 	"math"
+	"math/big"
+	"strings"
+	"time"
 
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/teamnsrg/go-ethereum/log"
 	"github.com/teamnsrg/go-ethereum/p2p/discover"
-	"math/big"
-	"time"
 )
 
 func (srv *Server) initSql() error {
-	if srv.MySQLName != "" {
+	if srv.MySQLName == "" {
+		log.Trace("No sql db connection info provided")
+	} else {
 		db, err := sql.Open("mysql", srv.MySQLName)
 		if err != nil {
 			log.Error("Failed to open sql db handle", "database", srv.MySQLName, "err", err)
 			return err
 		}
 		log.Trace("Opened sql db handle", "database", srv.MySQLName)
-		err = db.Ping()
-		if err != nil {
+		if err := db.Ping(); err != nil {
 			log.Error("Sql db connection failed ping test", "database", srv.MySQLName, "err", err)
-			if err := db.Close(); err != nil {
-				log.Error("Failed to close sql db handle", "database", srv.MySQLName, "err", err)
-			} else {
-				log.Trace("Closed sql db handle", "database", srv.MySQLName)
-			}
 			return err
 		}
 		log.Trace("Sql db connection passed ping test", "database", srv.MySQLName)
 		srv.DB = db
 
+		// backup tables
+		// If ResetSQL is true, BackupSQL should be true as well
+		// check both just in case
+		if srv.BackupSQL || srv.ResetSQL {
+			sqlNameParsed := strings.Split(srv.MySQLName, "/")
```

Comment:
Why does ResetSQL also trigger a backup? Shouldn't this be handled by the fact that you set both cfg.ResetSQL = true and cfg.BackupSQL = true when the user specifies the resetSqlFlag?

Reply:
Just an unnecessary sanity check lol...
I will remove it.